src/cpp/src/generation_config.cpp

```diff
@@ -131,6 +131,9 @@ bool GenerationConfig::is_speculative_decoding() const {
 }
 
 void GenerationConfig::validate() const {
+    OPENVINO_ASSERT(eos_token_id == -1 || stop_token_ids.find(eos_token_id) != stop_token_ids.end(),
+        "'stop_token_ids' must contain 'eos_token_id'. Please, call 'set_eos_token_id' with 'eos_token_id' value");
```

Comment:
you also need to port the fix of this test
https://github.com/openvinotoolkit/openvino.genai/pull/1319/files#diff-2a245797c6f7f6fd6dd3eeb7a2dff57cd2bd1a905e063605d766328428b99e8dR857